src/librustc_metadata/rmeta/encoder.rs

```diff
@@ -693,16 +693,23 @@ impl EncodeContext<'tcx> {
         vis: &hir::Visibility<'_>,
     ) {
         let tcx = self.tcx;
-        let def_id = tcx.hir().local_def_id(id).to_def_id();
+        let def_id = tcx.hir().local_def_id(id);
         debug!("EncodeContext::encode_info_for_mod({:?})", def_id);
 
         let data = ModData {
             reexports: match tcx.module_exports(def_id) {
-                Some(exports) => self.lazy(exports),
```

Comment:
Why didn't this "just work"?
`[LocalDefId]` should be a sequence of encodable things as well.

Reply:
This is the error i get i just call `self.lazy(exports)`:

```
error[E0277]: the trait bound `&rustc_middle::hir::exports::Export<rustc_span::def_id::LocalDefId>: rmeta::encoder::EncodeContentsForLazy<rustc_middle::hir::exports::Export<rustc_hir::hir_id::HirId>>` is not satisfied
   --> src/librustc_metadata/rmeta/encoder.rs:701:39
    |
701 |                 Some(exports) => self.lazy(exports),
    |                                       ^^^^ the trait `rmeta::encoder::EncodeContentsForLazy<rustc_middle::hir::exports::Export<rustc_hir::hir_id::HirId>>` is not implemented for `&rustc_middle::hir::exports::Export<rustc_span::def_id::LocalDefId>`
    |
    = note: required because of the requirements on the impl of `rmeta::encoder::EncodeContentsForLazy<[rustc_middle::hir::exports::Export<rustc_hir::hir_id::HirId>]>` for `&[rustc_middle::hir::exports::Export<rustc_span::def_id::LocalDefId>]`
```

Reply:
Oh I see i have to change `ModData` to store `LocalDefId`.

Reply:
So @petrochenkov it is really an invasive change to serialize/deserialize `LocalDefId` here. I am not sure it should part of this PR.

Reply:
Ok.